### PR TITLE
feat(patterns): Start migrating patterns from @agoric/store to new @endo/patterns

### PR DESCRIPTION
**Pull request by erights**

### CODE REVIEW

File: packages/patterns/package.json [hunk 1]
```diff
@@ -32,21 +28,25 @@
...
+    "eslint": "^7.32.0",
     "eslint-config-airbnb-base": "^14.0.0",
     "eslint-config-prettier": "^6.9.0",
     "eslint-plugin-eslint-comments": "^3.1.2",
-    "eslint-plugin-import": "^2.19.1",
+    "eslint-plugin-import": "^2.26.0",
     "eslint-plugin-prettier": "^3.4.1",
     "prettier": "^2.8.0",
-    "typescript": "~4.8.4"
+    "typescript": "~4.9.4"
```
Comment (erights) [inline packages/patterns/package.json:49]: @kriskowal 
Thanks for https://www.youtube.com/watch?v=SCOYsE5G7Xk. It is how I found out how to update the version numbers, and is what clued me in to put this tilda back in afterwards.

File: packages/patterns/package.json [hunk 2]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "@endo/patterns",
   "version": "0.1.0",
-  "private": null,
```
Comment (erights) [inline packages/patterns/package.json]: Please look after commit 1 to see the changes I made after repackage. Did I delete exactly what I should have deleted?

**Comment by erights** [comment]
Attn @Tyrosine22

**Comment by erights** [comment]
Aside from the one mystery at #1450 , it was amazingly painless to get to CI green! Yeah, repackage!

**Comment by FUDCo** [comment]
Since this is a cross-repo move, the git diffs make it entirely unclear which things are simply relocations and which are changes and/or additions.  I presume it's the latter that are of interest in reviewing?

**Comment by erights** [comment]
> Since this is a cross-repo move, the git diffs make it entirely unclear which things are simply relocations and which are changes and/or additions. I presume it's the latter that are of interest in reviewing?

At this point, my intention is to make it as blindly imitative as possible while still having the resulting package be a good endo citizen. This is almost all 
   * deleting things not relevant to the pattern portion of the old `@agoric/store` package
   * repairing imports
   * copying some functions from `@agoric/internal` utils.js to the new `utils.js`, until they can find their proper home.

The attached .patch file is from diffing the two and manually removing everything that obviously does not need to be looked at. Look at in an editor that's friendly for looking at diffs.

[pattern-migration.patch](https://github.com/endojs/endo/files/10441557/pattern-migration.patch)

**Comment by erights** [comment]
From diffing only the respective utils.js files. As you can see, it is all deletions aside from 
   * `identChecker` which is copied literally from `@agoric/assert`
   * import repair
   * #1450 repair.

[utils.patch](https://github.com/endojs/endo/files/10441623/utils.patch)

File: packages/patterns/package.json (hunk 2 above)
Comment (kriskowal) [reply packages/patterns/package.json]: All good.

**Review by kriskowal — APPROVED** [review]
Trusting that the code is largely the same as what integrated in Agoric SDK, this seems fine. So you have a PR coming to refactor Agoric SDK in place to use these?

**Comment by erights** [comment]
> So you have a PR coming to refactor Agoric SDK in place to use these?

I will.

File: packages/patterns/package.json [hunk 3]
```diff
@@ -0,0 +1,77 @@
+{
+  "name": "@endo/patterns",
+  "version": "0.1.0",
+  "description": "Description forthcoming.",
```
Comment (turadg) [inline packages/patterns/package.json:4]: @erights looks like this is still needed

File: packages/patterns/package.json [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  "keywords": [],
+  "author": "Endo contributors",
+  "license": "Apache-2.0",
+  "homepage": "https://github.com/endojs/endo/tree/master/packages/patterns#readme",
```
Comment (turadg) [inline packages/patterns/package.json:8]: the fragment only works for projects, not files in the tree.

should be: https://github.com/endojs/endo/blob/master/packages/patterns/README.md

Incidentally, it's empty and needs contents.

Comment (erights) [reply packages/patterns/package.json:8]: See https://github.com/endojs/endo/pull/1480

(Separately, still needs contents)

Comment (erights) [reply packages/patterns/package.json:8]: The empty READMEs for pass-style, patterns, and exo will be minimally fixed by https://github.com/endojs/endo/pull/1480 with initial text that's much better than nothing. A first step.

**Pull request merged**